### PR TITLE
Add syntax highlighting for raw JS code

### PR DESCRIPTION
**Pull request by fhammerschmidt**
image

image

implements https://github.com/rescript-lang/rescript-compiler/issues/6214

### CODE REVIEW

**Comment by fhammerschmidt** [comment]
Note: I did not implement it for `""` strings, because I think that would break too easily.

**Review by zth — APPROVED** [review]
Nice! Mind adding to the changelog?

**Comment by cristianoc** [comment]
Actually would you do it for `%ffi`?

**Comment by zth** [comment]
> Actually would you do it for `%ffi`? Notice single `%`

Judging from the screenshot this already works, no?

**Comment by cristianoc** [comment]
> > Actually would you do it for `%ffi`? Notice single `%`
> 
> 
> 
> Judging from the screenshot this already works, no?

Oh right I did not see the second one.

File: grammars/rescript.tmLanguage.json [hunk 1]
```diff
@@ -481,12 +481,41 @@
...
+        }
+      ]
     }
   },
   "patterns": [
     {
       "include": "#storage"
     },
+    {
+      "include": "#ffi"
```
Comment (cristianoc) [inline grammars/rescript.tmLanguage.json:517]: Should we alphabetize these?

**Review by cristianoc — APPROVED** [review]
This is great thanks!

File: grammars/rescript.tmLanguage.json (hunk 1 above)
Comment (fhammerschmidt) [reply grammars/rescript.tmLanguage.json:517]: I guess it does not hurt. Will do it.

**Review by cristianoc — APPROVED** [review]

**Pull request merged**